equinix/resource_fabric_service_profile_acc_test.go

```diff
@@ -17,139 +17,135 @@ import (
 	v4 "github.com/equinix-labs/fabric-go/fabric/v4"
 )
 
-func testAccFabricReadServiceProfileConfig(uuid string) string {
-	return fmt.Sprintf(`data "equinix_fabric_service_profile" "test" {
-	uuid = "%s"
-	}
-`, uuid)
-}
+func TestAccFabricCreateServiceProfile_PFCR(t *testing.T) {
+	ports := GetFabricEnvPorts(t)
 
-func testAccFabricReadServiceProfilesListConfig(name string) string {
-	return fmt.Sprintf(`data "equinix_fabric_service_profiles" "test" {
-		filter {
-			property = "/name"
-			operator = "="
-			values = ["%s"]
-		}
+	var portTcFirst, portTcSecond v4.Port
+	if len(ports) > 0 {
+		portTcFirst = ports["pfcr"]["dot1q"][0]
+		portTcSecond = ports["pfcr"]["qinq"][0]
 	}
-`, name)
-}
 
-func TestAccFabricReadServiceProfileByUuid(t *testing.T) {
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:  func() { acceptance.TestAccPreCheck(t) },
-		Providers: acceptance.TestAccProviders,
+		PreCheck:     func() { acceptance.TestAccPreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
+		CheckDestroy: checkServiceProfileDelete,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccFabricReadServiceProfileConfig("bfb74121-7e2c-4f74-99b3-69cdafb03b41"),
+				Config: testAccFabricCreateServiceProfileConfig(portTcFirst.Uuid, "XF_PORT", portTcFirst.Location.MetroCode),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(
-						"data.equinix_fabric_service_profile.test", "name", fmt.Sprint("Azure ExpressRoute")),
+						"equinix_fabric_service_profile.test", "name", fmt.Sprint("ds_con_sp_PFCR")),
 					resource.TestCheckResourceAttr(
-						"data.equinix_fabric_service_profile.test", "uuid", fmt.Sprint("bfb74121-7e2c-4f74-99b3-69cdafb03b41")),
-				),
-			},
-		},
-	})
-}
-
-func TestAccFabricSearchServiceProfilesByName(t *testing.T) {
-	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:  func() { acceptance.TestAccPreCheck(t) },
-		Providers: acceptance.TestAccProviders,
-		Steps: []resource.TestStep{
-			{
-				Config: testAccFabricReadServiceProfilesListConfig("Azure ExpressRoute"),
-				Check: resource.ComposeTestCheckFunc(
+						"equinix_fabric_service_profile.test", "type", fmt.Sprint("L2_PROFILE")),
 					resource.TestCheckResourceAttr(
-						"data.equinix_fabric_service_profiles.test", "data.#", fmt.Sprint(1)), // Check  total number of ServiceProfile list returned in the Response payloads
+						"equinix_fabric_service_profile.test", "state", fmt.Sprint("ACTIVE")),
```

Comment:
There's no need to use `fmt.Sprint()` for your strings. You can just use "" for your string values.

Reply:
This will need to be updated for all `fmt.Sprint()`'s that just have a simple string primitive inside them.

Reply:
done